File: frontend/src/types/competition.ts
export interface CompetitionBusinessMaterial {
  id: string
  filename: string
  uploaded_at: string
  status: string
}

export interface CompetitionBusinessFact {
  id: string
  label: string
  value: string
  status: string
}

export interface CompetitionBusinessSupplement {
  id: string
  title: string
  why: string
  accepted_materials: string[]
  status: string
}

export interface CompetitionBusinessProgress {
  label: string
  state: 'completed' | 'current' | 'upcoming'
}

export interface CompetitionBusinessCenter {
  scenario_id: number
  project_name: string
  materials: CompetitionBusinessMaterial[]
  facts: CompetitionBusinessFact[]
  supplements: CompetitionBusinessSupplement[]
  progress: CompetitionBusinessProgress[]
  current_status: string
}

File: backend/app/schemas/competition.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


class CompetitionBusinessMaterial(BaseModel):
    id: str
    filename: str
    uploaded_at: datetime
    status: str


class CompetitionBusinessFact(BaseModel):
    id: str
    label: str
    value: str
    status: str


class CompetitionBusinessSupplement(BaseModel):
    id: str
    title: str
    why: str
    accepted_materials: list[str]
    status: str


class CompetitionBusinessProgress(BaseModel):
    label: str
    state: Literal["completed", "current", "upcoming"]


class CompetitionBusinessCenterResponse(BaseModel):
    scenario_id: int
    project_name: str
    materials: list[CompetitionBusinessMaterial]
    facts: list[CompetitionBusinessFact]
    supplements: list[CompetitionBusinessSupplement]
    progress: list[CompetitionBusinessProgress]
    current_status: str


class CompetitionBusinessFactConfirmRequest(BaseModel):
    value: str = Field(min_length=1, max_length=10_000)

    model_config = {"extra": "forbid"}
